everett/manager.py

```diff
@@ -240,6 +241,100 @@ def __repr__(self):
         return '<ListOf(%s)>' % qualname(self.sub_parser)
 
 
+def _parse_rwx(modestr, default):
+    """parse out 'r', 'w' and 'x'.  use 'default' if none used or specified"""
+    mode = 0
+    if 'r' in modestr:
+        mode |= os.R_OK
+    if 'w' in modestr:
+        mode |= os.W_OK
+    if 'x' in modestr:
+        mode |= os.X_OK
+    if not mode:
+        mode = default
+    return mode
+
+
+def _mode2english(mode):
+    """turn a file mode into an english phrase"""
+    perms = []
+    if mode & os.R_OK: perms.append('read')
+    if mode & os.W_OK: perms.append('write')
+    if mode & os.X_OK: perms.append('execute')
+    return (','.join(perms[:-1]) + ' and ' if len(perms)>1 else '') + perms[-1]
+
+
+class PathValue(object):
+    """
+    A parser for paths
+
+    config('Pathname', parser=PathValue('w', isdir=True, exists=False))
+    """
+
+    def __init__(self, mode='', isdir=None, exists=True):
+        """
+        :arg mode: specifies if the file is any subset of readable ('r'), writable ('w') and executable ('x')
+        Ignores any other characters in 'mode'; defaults to 'r' if mode is unspecified, or 'rx' if isdir is True
+
+        :arg isdir: if True, check that it's a directory and not just any kind of file
+
+        :arg exists: if True, verify that the path exists.
+        """
+        self.isdir = isdir
+        self.exists = exists
+        self.mode = _parse_rwx(mode, os.R_OK|os.X_OK if isdir else os.R_OK)
+
+    def __call__(self, val):
+        """
+        :returns: (parser that returns) a normalized absolute path to the configured file
+        """
+        if self.exists and not os.path.exists(val):
+            raise ValueError("Expected %s to exist, but it doesn't" % val)
+        if not os.access(val, self.mode):
+            raise ValueError('Unable to %s %s' % (_mode2english(self.mode), val))
+        if self.isdir is not None:
```

Comment:
When would `isdir` be `None`?

Reply:
Oh, I see what's going on.

There are other kinds of things in a file system than directories and files. This code seems to suggest there's just one or the other.

What do other libs do for an "make sure this is a directory" check?

Reply:
The code checks if `.isdir`:
  * `True`: must be a directory
  * `False`: must be a file (per `os.path.isfile()`, which follows symlinks, but will fail on, say, character devices or named pipes)
  * `None`: don't check

I don't know what other libraries to compare to, but I'm happy to dig up details on any you care to name.